File: src/Shared/Logistics.Shared.Models/Reports/FinancialsReportDto.cs
using Logistics.Domain.Primitives.Enums;

namespace Logistics.Shared.Models;

public record FinancialsReportDto 
{
    public decimal TotalInvoiced { get; set; }
    public decimal TotalPaid { get; set; }
    public decimal TotalDue { get; set; }

    public int FullyPaidInvoices { get; set; }
    public int PartiallyPaidInvoices { get; set; }
    public int UnpaidInvoices { get; set; }
    public decimal AverageInvoiceValue { get; set; }
    public decimal CollectionRate { get; set; }

    public List<PaymentTrendDto> PaymentTrends { get; set; }
    public List<TopCustomerDto> TopCustomers { get; set; }
    public StatusDistributionDto StatusDistribution { get; set; }
}

public class PaymentTrendDto
{
    public string Period { get; set; }
    public decimal Invoiced { get; set; }
    public decimal Paid { get; set; }
    public double CollectionRate { get; set; }
}

public class TopCustomerDto
{
    public string CustomerName { get; set; }
    public decimal TotalInvoiced { get; set; }
    public decimal TotalPaid { get; set; }
    public int InvoiceCount { get; set; }
}

public class StatusDistributionDto
{
    public double PaidPercentage { get; set; }
    public double PartialPercentage { get; set; }
    public double UnpaidPercentage { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/api/models/report/financials-report.dto.ts
export interface FinancialsReportDto {
  totalPaid: number;
  totalDue: number;
  totalInvoiced: number;
  fullyPaidInvoices: number;
  partiallyPaidInvoices: number;
  unpaidInvoices: number;
  averageInvoiceValue: number;
  collectionRate: number;
  outstandingAmount: number;
  overdueAmount: number;
  overdueInvoices: number;
  paymentTrends: PaymentTrendDto[];
  topCustomers: TopCustomerDto[];
  statusDistribution: StatusDistributionDto;
  revenueTrends: RevenueTrendDto[];
  financialMetrics: FinancialMetricDto[];
}

export interface PaymentTrendDto {
  period: string;
  invoiced: number;
  paid: number;
  collectionRate: number;
}

export interface TopCustomerDto {
  customerName: string;
  totalInvoiced: number;
  totalPaid: number;
  invoiceCount: number;
}

export interface StatusDistributionDto {
  paidPercentage: number;
  partialPercentage: number;
  unpaidPercentage: number;
}

export interface RevenueTrendDto {
  period: string;
  revenue: number;
  profit: number;
  expenses: number;
  profitMargin: number;
}

export interface FinancialMetricDto {
  metric: string;
  value: number;
  unit: string;
  trend: number;
  category: string;
}
